kolibri/plugins/base.py

```diff
@@ -78,3 +83,69 @@ def enable(cls):
     def disable(cls):
         """Modify the kolibri config dict to your plugin's needs"""
         cls._installed_apps_remove()
+
+
+class KolibriFrontEndPluginBase(KolibriPluginBase):
+    """
+    This is the class that all plugins that wish to load any assets into the front end
+    must implement, in order for them to be part of the webpack asset loading pipeline.
+    Minimally these must implement the following properties and methods:
+
+    The name of the frontend plugin.
+    name = "example_plugin"
+
+    The path to the Javascript file that defines the plugin/acts as the entry point.
+    entry_file = "assets/js/example_plugin.js"
+
+    This hook will register the frontend plugin to be available for rendering its built files into Django templates.
+    def hooks(self):
+        return {
+            FRONTEND_PLUGINS: self._register_front_end_plugins
+        }
+    """
+
+    @classmethod
+    def webpack_bundle_data(cls):
+        """
+        Returns information needed by the webpack parsing process.
+        :return: dict with keys "name", "entry_file", and, "external".
+        "name" - is the name that the frontend plugin has.
+        "entry_file" - is the Javascript file that defines the plugin.
+        "external" - an optional flag used only by the kolibri_core plugin.
+        """
+        try:
+            return {
+                "name": cls.name,
```

Comment:
I see. So all subclasses must now implement the following class functions and attributes?